hnn_core/feed.py

```diff
@@ -36,251 +70,207 @@ class ExtFeed(object):
-                self.seed2 = self.params['prng_seedcore']
-            else:
-                self.seed = self.params['prng_seedcore'] + self.gid
-        else:  # if seed explicitly specified use it
-            self.seed = seed
-            if hasattr(self, 'seed2'):
-                self.seed2 = seed
-        self.prng = np.random.RandomState(self.seed)
-        if hasattr(self, 'seed2'):
-            self.prng2 = np.random.RandomState(self.seed2)
-
-    def set_event_times(self):
-
-        # check feed name validity, allowing substring matches ('evprox1' etc)
-        valid_feeds = ['extpois', 'extgauss', 'common', 'evprox', 'evdist']
-        # NB check if self.feed_type has a valid substring, not vice versa
-        matches = [f for f in valid_feeds if f in self.feed_type]
-        if len(matches) == 0:
-            raise ValueError('Invalid external feed: %s' % self.feed_type)
-        elif len(matches) > 1:
-            raise ValueError('Ambiguous external feed: %s' % self.feed_type)
-
-        # Each of these methods creates self.event_times
-        # Return values not checked: False if all weights for given feed type
-        # are zero. Designed to be silent so that zeroing input weights
-        # effectively disables each.
-        if self.feed_type == 'extpois':
-            self._create_extpois()
-        elif self.feed_type.startswith(('evprox', 'evdist')):
-            self._create_evoked()
-        elif self.feed_type == 'extgauss':
-            self._create_extgauss()
-        elif self.feed_type == 'common':
-            self._create_common_input()
-
-    # based on cdf for exp wait time distribution from unif [0, 1)
-    # returns in ms based on lamtha in Hz
-    def _t_wait(self, lamtha):
-        return -1000. * np.log(1. - self.prng.rand()) / lamtha
-
-    # new external pois designation
-    def _create_extpois(self):
-        # XXX Check whether both AMPA and NMDA connections are zero
-        # If so, return without updating self.event_times (from empty list)
-        if self.params[self.cell_type][0] <= 0.0 and \
-                self.params[self.cell_type][1] <= 0.0:
-            return False  # 0 ampa and 0 nmda weight
-        # check the t interval
-        t0 = self.params['t_interval'][0]
-        T = self.params['t_interval'][1]
-        if t0 < 0:
-            raise ValueError('The start time for Poisson inputs must be'
-                             f'greater than 0. Got {t0}')
-        if T < t0:
-            raise ValueError('The end time for Poisson inputs must be'
-                             f'greater than start time. Got ({t0}, {T})')
-        lamtha = self.params[self.cell_type][3]  # ind 3 is frequency (lamtha)
-        # values MUST be sorted for VecStim()!
-        # start the initial value
-        val_pois = np.array([])
-        if lamtha > 0.:
-            t_gen = t0 + self._t_wait(lamtha)
-            if t_gen < T:
-                np.append(val_pois, t_gen)
```

Comment:
@rythorpe found the source of the discrepancy you noticed in the gamma example. When you use [`np.append`](https://numpy.org/doc/stable/reference/generated/numpy.append.html), the return value needs to be `val_pois`. The array is not modified in-place like a  list. So, this is actually a bug in the old code. One spike is missed at the beginning. Should we consider this a bugfix or should I push a commit to revert to the old behavior?

Reply:
Oh I see. This is a good find. My vote is to move forward with your current solution as a bug fix and document it in `whats_new.rst`.